app/controllers/users/sessions_controller.rb

```diff
@@ -113,7 +121,7 @@ def update_sp_return_logs_with_user(user_id)
     end
 
     def expires_at
-      @_expires_at ||= (session[:session_expires_at]&.to_datetime || (now - 1))
+      session[:session_expires_at]&.to_datetime || (now - 1)
```

Comment:
Since this gets updated now, the memoization potentially causes issues. I don't think the performance hit should be significant.